lib/rentlinx/errors.rb

```diff
@@ -7,4 +7,37 @@ class UnexpectedAttributes < RentlinxError
 
   class InvalidTypeParam < RentlinxError
   end
+
+  class InvalidObject < RentlinxError
+    def initialize(type)
+      super("The given #{type} is missing required attributes and cannot be posted.")
+    end
+  end
+
+  class HTTPError < RentlinxError
+  end
+
+  class BadRequest < HTTPError
+    def initialize
+      super('The request sent to the server was invalid.')
+    end
+  end
+
+  class NotFound < HTTPError
+    def initialize
+      super('The item you requested could not be found on the remote server.')
+    end
+  end
+
+  class Forbidden < HTTPError
+    def initialize
+      super('You are not permitted to access the item you requested.')
```

Comment:
Type might be helpful here as well (https://github.com/appfolio/rentlinx_client/pull/15/files#r33316372)